lib/contentTypeParser.js

```diff
@@ -195,48 +195,44 @@ ContentTypeParser.prototype.run = function (contentType, handler, request, reply
       parser,
       done
     )
-  } else {
-    const result = parser.fn(request, request[kRequestPayloadStream], done)
+    return
+  }
 
-    if (result && typeof result.then === 'function') {
-      result.then(body => done(null, body), done)
-    }
+  const result = parser.fn(request, request[kRequestPayloadStream], done)
+  if (result && typeof result.then === 'function') {
```

Comment:
`nit`; it performs better compared to:
```suggestion
  if (typeof result?.then === 'function') {
```

?